src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -3122,6 +3133,71 @@ NormalizeMultiRowInsertTargetList(Query *query)
 }
 
 
+/*
+ * GetNumberOfUnspecifiedDefaultCols returns number of default columns that
+ * are not explicitly specified in given INSERT query's values_lists. We do
+ * that by counting non-Var nodes in targetList of given query object.
+ */
+static int
+GetNumberOfUnspecifiedDefaultCols(Query *query)
+{
+	int unspecifiedDefaultCols = 0;
+
+	TargetEntry *targetEntry = NULL;
+	foreach_ptr(targetEntry, query->targetList)
+	{
+		Expr *targetExpr = targetEntry->expr;
+		if (!IsA(targetExpr, Var))
+		{
+			unspecifiedDefaultCols++;
+		}
+	}
+
+	return unspecifiedDefaultCols;
+}
+
+
+/*
+ * AppendUnspecifiedDefaultColReferences appends dummy reference names for each
+ * column that has a DEFAULT definition but not specified in VALUES clause of
+ * INSERT query.
+ */
+static void
+AppendUnspecifiedDefaultColReferences(RangeTblEntry *valuesRTE,
+									  int unspecifiedDefaultCols)
+{
+	/*
+	 * Postgres starts from 1 to name columns in VALUES clause. So we start
+	 * with next integer to naming columns that we explicitly append to target
+	 * list in NormalizeMultiRowInsertTargetList function.
+	 * Also see addRangeTableEntryForValues.
+	 */
+	int existingColReferences = list_length(valuesRTE->eref->colnames);
+	int unspecifiedColId = existingColReferences + 1;
+	for (int i = 0; i < unspecifiedDefaultCols; i++)
+	{
+		Value *missingColumnString = MakeColumnString(unspecifiedColId++);
+		valuesRTE->eref->colnames = lappend(valuesRTE->eref->colnames,
+											missingColumnString);
+	}
+}
+
+
+/*
+ * MakeColumnString returns a String (Value) object by appending given integer
+ * to end of the "column" string.
+ */
+static Value *
+MakeColumnString(int i)
+{
+	StringInfo missingColStrInfo = makeStringInfo();
+	appendStringInfo(missingColStrInfo, "column%d", i);
```

Comment:
what if a column named `column1`  exists?

Reply:
as far as I see in `addRangeTableEntryForValues`, it doesn't seem possible actually https://github.com/citusdata/citus/blob/a81ebb260aed9f2516eefe7b21aaf638a7fe4944/src/backend/distributed/planner/multi_router_planner.c#L3165-L3174